File: codes/jiheon/121_besttimetobuyandsellstock.java
class Solution {
     public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int answer = 0;
        for (var price:prices){
            if (minPrice > price){
                minPrice = price;
            }else{
                answer = Math.max(answer,price-minPrice);
            }
        }
        return answer;
    }
}

File: codes/jiheon/121_besttimetobuyandsellstock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var answer = 0;
        var minPrice = Int.MAX_VALUE;
        for (price in prices){
               if (price< minPrice) minPrice = price
               else answer = max(answer,price-minPrice);
        }
        return answer
    }
}